tests/tests/test_autogenerate_delta.py

```diff
@@ -0,0 +1,155 @@
+    ):
+        """Upgrade a device with auto-generated delta artifact"""
+
+        mender_device = env.device
+        devauth = DeviceAuthV2(env.auth)
+        deploy = Deployments(env.auth, devauth)
+
+        mender_conf = mender_device.run("cat /etc/mender/mender.conf")
+        valid_image_rofs = valid_image_rofs_commercial_with_mender_conf(mender_conf)
+        if valid_image_rofs is None:
+            pytest.skip(
+                "Thud branch and older from Yocto does not have R/O rootfs support"
+            )
+
+        # Configure delta auto-generation
+        dc = ApiClient(
+            deployments.URL_INTERNAL,
+            host=get_container_manager().get_ip_of_service("mender-deployments")[0]
+            + ":8080",
+            schema="http://",
+        )
+
+        r = dc.call(
+            "PUT",
+            deployments.URL_INTERNAL_CONFIG,
+            body={
+                "delta": {
+                    "enabled": True,
+                    "binary_delta_limits": {
+                        "jobs_in_parallel": {"max": 2},
+                        "queue_length": {"max": 4},
+                    },
+                    "binary_delta": {"timeout": 3600},
+                }
+            },
+            path_params={"tenant_id": env.tenant["id"]},
+        )
+
+        assert r.ok, "Failed to set the delta configuration"
+
+        def make_artifact(artifact_file, artifact_id):
+            return image.make_rootfs_artifact(
+                valid_image_rofs,
+                conftest.machine_name,
+                artifact_id,
+                artifact_file,
+                provides={
+                    "rootfs-image.update-module.mender-binary-delta.mender_update_module": "mender-binary-delta",
+                    "rootfs-image.update-module.mender-binary-delta.version": "mender-image-master",
+                },
+            )
+
+        host_ip = env.get_virtual_network_host_ip()
+        update_image(
+            mender_device,
+            host_ip,
+            make_artifact=make_artifact,
+            devauth=devauth,
+            deploy=deploy,
+        )
+
+        def make_artifact2(artifact_file, artifact_id):
+            return image.make_rootfs_artifact(
+                valid_image_rofs,
+                conftest.machine_name,
+                artifact_id,
+                artifact_file,
+                provides={
+                    "rootfs-image.update-module.mender-binary-delta.mender_update_module": "mender-binary-delta2",
+                    "rootfs-image.update-module.mender-binary-delta.version": "mender-image-master2",
+                },
+            )
+
+        # check if client is applying delta
+        isDelta = False
+
+        def deployment_triggered_callback():
+            # wait a minute
+            i = 60
+            while i > 0:
+                output = mender_device.run(
+                    "systemctl status --no-pager -l -n 1000000 mender-client | grep 'xdelta3'",
```

Comment:
A bit flaky as a detection mechanism, however as we know we cannot use the Server's APIs to get this information just yet. However, once https://tracker.mender.io/browse/MEN-6138 is done, can't we do it that way?

Reply:
yes, I don't like this part too, and this is why I was trying to use backend API; I can see the MEN-6138 is in progress so I can wait and update the test (probably today)

Reply:
or I'll just follow up with that; I'll have to run pipeline once again anyway